factory/tests/tests_basics.rs

```diff
@@ -0,0 +1,192 @@
+        .call(contract.id(), "deploy_new_auction")
+        .args_json(
+            json!({"name": "new-auction", "end_time": a_minute_from_now.to_string(),"auctioneer": auctioneer.id(),"ft_contract": ft_contract.id(),"nft_contract": nft_account.id(),"token_id":"1", "starting_price":starting_price }),
+        )
+        .max_gas()
+        .deposit(NearToken::from_millinear(1600))
+        .transact()
+        .await?;
+
+    assert!(deploy_new_auction.is_success());
+
+    let auction_account_id: AccountId = format!("new-auction.{}", contract.id())
+        .parse()
+        .unwrap();
+
+    // Register accounts
+    for account_id in [
+        alice.id().clone(),
+        auction_account_id.clone(),
+    ]
+    .iter()
+    {
+        let register = ft_contract
+            .call("storage_deposit")
+            .args_json(serde_json::json!({ "account_id": account_id }))
+            .deposit(NearToken::from_yoctonear(8000000000000000000000))
+            .transact()
+            .await?;
+    
+        assert!(register.is_success());
+    }
+
+    // Transfer FTs
+    let transfer_amount = U128(150_000);
+
+    let root_transfer_alice =
+        ft_transfer(&root, alice.clone(), ft_contract.clone(), transfer_amount).await?;
+    assert!(root_transfer_alice.is_success());
+
+    // Alice makes a bid
+    let alice_bid = ft_transfer_call(
+        alice.clone(),
+        ft_contract.id(),
+        &auction_account_id,
+        U128(50_000),
+    )
+    .await?;
+
+    assert!(alice_bid.is_success());
+
+    let highest_bid_alice: Bid = alice.view(&auction_account_id, "get_highest_bid").args_json({}).await?.json()?;
+    assert_eq!(highest_bid_alice.bid, U128(50_000));
+    assert_eq!(highest_bid_alice.bidder, *alice.id());
+
+    let contract_account_balance: U128 = ft_balance_of(&ft_contract, &auction_account_id).await?;
+    assert_eq!(contract_account_balance, U128(50_000));
+    let alice_balance_after_bid: U128 = ft_balance_of(&ft_contract, alice.id()).await?;
+    assert_eq!(alice_balance_after_bid, U128(100_000));
+
+    // Try to launch a new auction with insufficient deposit
+    let deploy_new_auction: ExecutionFinalResult = alice
+        .call(contract.id(), "deploy_new_auction")
+        .args_json(
+            json!({"name": "new-auction", "end_time": a_minute_from_now.to_string(),"auctioneer": auctioneer.id(),"ft_contract": ft_contract.id(),"nft_contract": nft_account.id(),"token_id":"1", "starting_price":starting_price }),
+        )
+        .max_gas()
+        .deposit(NearToken::from_millinear(1400))
+        .transact()
+        .await?;
+
+    assert!(deploy_new_auction.is_failure());
+
+    Ok(())
+}
+
+async fn create_subaccount(
+    root: &near_workspaces::Account,
+    name: &str,
+) -> Result<near_workspaces::Account, Box<dyn std::error::Error>> {
+    let subaccount = root
+        .create_subaccount(name)
+        .initial_balance(TEN_NEAR)
+        .transact()
+        .await?
+        .unwrap();
```

Comment:
I love the idea of moving such big function chains in separate methods.

I will apply the same approach to my code